eth/gasprice/gasprice.go

```diff
@@ -249,10 +270,24 @@ func (oracle *Oracle) SuggestPrice(ctx context.Context) (*big.Int, error) {
 		return nil, err
 	}
 
+	// Fetch the most recent block by number
+	block, err := oracle.backend.BlockByNumber(ctx, rpc.LatestBlockNumber)
+	if err != nil {
+		return nil, err
+	}
+
+	// If the block does have a baseFee, calculate the next base fee
+	// based on the current time and add it to the tip to estimate the
+	// total gas price estimate.
+	feeConfig, err := oracle.backend.GetFeeConfigAt(block.Header())
+	if err != nil {
+		return nil, err
+	}
+
 	// We calculate the [nextBaseFee] if a block were to be produced immediately.
 	// If [nextBaseFee] is lower than the estimate from sampling, then we return it
 	// to prevent returning an incorrectly high fee when the network is quiescent.
-	nextBaseFee, err := oracle.estimateNextBaseFee(ctx)
+	nextBaseFee, err := oracle.estimateNextBaseFee(block, feeConfig)
```

Comment:
In both places we call `estimateNextBaseFee` we fetch the latest block and the fee config at that block, so we should move both of these calls into `estimateNextBaseFee` and change from passing in these two params to passing in the `ctx`.